cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/DatastoreWrapper.java

```diff
@@ -88,6 +90,25 @@ private String getCheckpointDocumentId(String replicatorIdentifier) {
         return "_local/" + replicatorIdentifier;
     }
 
+    public void bulkInsert(List<BasicPullStrategy.BatchItem> batches, boolean pullAttachmentsInline) throws DocumentException  {
+        List<ForceInsertItem> itemsToInsert = new ArrayList<ForceInsertItem>();
+        for (BasicPullStrategy.BatchItem batch : batches) {
```

Comment:
Minor point as I doubt this `ArrayList` will ever be huge enough to make a massive performance difference, but as this is a performance PR I'll mention it anyway - traditional `for` loops are more efficient than enhanced `for` loops for `ArrayList` (but not for most other collections), so it _might_ be worth giving it a try to see what difference it makes.  See:
http://developer.android.com/training/articles/perf-tips.html#Loops